### PR TITLE
Feature/fbot 500 option to open web chat links to the same window

### PR DESCRIPTION
**Pull request by janbadura**
Added option to have persistent conversation and opening links in the  same window

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -29,12 +29,28 @@ export const App = async (props: AppProps, container?: HTMLElement) => {
...
   };
 
+  if(body.config?.persistent === "user" || body.config?.persistent === "conversation" ){
+    if(sessionStorage.getItem("feedbotUserId")){
+      props.user.id = sessionStorage.getItem("feedbotUserId")
+    } else {
+      props.user.id = MakeId()
+      sessionStorage.setItem("feedbotUserId", props.user.id)
+    }
+
```
Comment (dvorakjan) [inline src/App.tsx:46]: Here set new App & Chat prop `openUrlTarget: [new, same, same-domain]` + pls add to README.md

File: src/App.tsx [hunk 2]
```diff
@@ -29,12 +29,28 @@ export const App = async (props: AppProps, container?: HTMLElement) => {
   konsole.log("BotChat.App props", props);
 
   // FEEDYOU generate user ID if not present in props, make sure its always string
+
```
Comment (dvorakjan) [inline src/App.tsx:32]:
```suggestion
```

File: src/App.tsx [hunk 3]
```diff
@@ -63,8 +79,10 @@ export const App = async (props: AppProps, container?: HTMLElement) => {
       const body = await response.json();
       console.log("Token response", body);
 
+      const dl = props.directLine || {}
```
Comment (dvorakjan) [inline src/App.tsx:82]:
```suggestion
      const directLine = props.directLine || {}
```

File: src/AdaptiveCardContainer.tsx [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { AjaxResponse, AjaxRequest } from 'rxjs/observable/dom/AjaxObservable';
 import * as adaptivecardsHostConfig from '../adaptivecards-hostconfig.json';
 import * as konsole from './Konsole';
 import { ChatState, AdaptiveCardsState } from './Store';
+import {ChangeUrlAction} from "./ChangeUrlAction"
```
Comment (dvorakjan) [inline src/AdaptiveCardContainer.tsx:12]: This `ChangeUrlAction` related stuff will probably won't be needed anymore.

**Review by dvorakjan — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -63,8 +68,26 @@ export const App = async (props: AppProps, container?: HTMLElement) => {
...
+      if(body.config.persist === "conversation"){
+        props.directLine.conversationId = sessionStorage["feedbotConversationId"]
+        props.directLine.webSocket = false
+      }
+
+      const directLine = props.directLine || {}
       props.botConnection = new DirectLine({
-        ...(props.directLine || {}),
+        ...directLine,
+        conversationId: directLine.conversationId || sessionStorage["feedbotConversationId"],
```
[the remark this answers is not in the document]
Comment (janbadura) [reply src/App.tsx:90]: It's there if ```directLine.conversationId``` is undefined which can happen depending on the mode which is used, then it uses sessionStorage.

**Pull request merged**